report/4.Solution/SemanticsDescription.tex

```diff
@@ -0,0 +1,748 @@
+\end{center}
+
+\texttt{[$TextToString_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, sto \vdash e \rightarrow_{e} v'}
+			{env_V, sto \vdash e\ toString \rightarrow_{t} v}
+	\end{math}
+	
+	\texttt{ where } $v = "v'"$
+\end{center}
+
+\textbf{[$TextLit_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	env_V,\ sto \vdash t \rightarrow_b v
+	\end{math}
+	\texttt{ if } $\mathbb{T}[t] = "v"$
+\end{center}
+
+\textbf{\Large{List Expressions}}\\
+List expressions returns a value $v$, where $v$ consists of a sequence of expression values separated with commas, encapsulated in $[\ ]$.
+$v$ is denoted $lst$, and is on the form $[(el \cup Lambda)~]$.
+The set of which it is an element of is denoted $Lit$.
+The transition-system for $list_{exp}$ is $(list_{exp} \cup List, \rightarrow_{list}, List \cup e)$.
+The transitions are shown below.
+The length transition returns a value $v$, where $v \in Z$.
+Even though this is an exception to the general description of a list expression, the transition has the same format as the rest of the transitions.
+
+\texttt{[$ListLength_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, sto \vdash num \rightarrow_{a} v}
+			{env_V, sto \vdash list\ length \rightarrow_{list} v}
+	\end{math}
+	
+	\texttt{ where } $num = |list|$
+\end{center}
+
+\texttt{[$ListAt_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+		\cfrac
+			{env_V, sto \vdash a \rightarrow_{a} v' \quad env_V, sto \vdash ls_v' \rightarrow_e v}
+			{env_V, sto \vdash list\ @\ a \rightarrow_{list} v}
+	\end{math}
+	
+	\texttt{ where } $ls_v'\ is\ the\ v'th\ element\ of\ list$
+\end{center}
+
+\textbf{[$ListLit_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	env_V,\ sto \vdash lst \rightarrow_{list} v
+	\end{math}
+	\texttt{ if } $\mathbb{LST}[lst] = v$
+\end{center}
+
+\textbf{\Large{Tuple Expressions}}\\
+Tuple expressions returns a value $v$, where $v$ consists of a sequence of expression values separated with commas, encapsulated in $()$.
+$v$ is denoted $tpl$, and is on the form $((el \cup Lambda)~)$.
+The set of which it is an element of is denoted $Tuple$.
+The transition-system for $tuple_{exp}$ is $(tuple_{tuple_exp} \cup Tuple, \rightarrow_{tuple}, Tuple)$.
+The transitions are shown below.
+
+\texttt{[$TplAt_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	\cfrac
+	{env_V, sto \vdash a \rightarrow_{a} v' \quad env_V, sto \vdash tp_v' \rightarrow_e v}
+	{env_V, sto \vdash tuple\ @\ a \rightarrow_{tuple} v}
+	\end{math}
+	
+	\texttt{ where } $tp_v'\ is\ the\ v'th\ element\ of\ tuple$
+\end{center}
+
+\textbf{[$ListLit_{BSS}$]}\\
+\begin{center}
+	\begin{math}
+	env_V,\ sto \vdash tpl \rightarrow_{tuple} v
+	\end{math}
+	\texttt{ if } $\mathbb{TPL}[tpl] = v$
+\end{center}
+
+\subsection{Big-Step Semantic for Statements}
+Statements $Stm$ can change the change the store, because they can change the value of an existing variable.
+The transition-system for $stm$ is $(Stm\ X\ Sto\ X \cup Sto, \rightarrow_{stm}, Sto)$.
```

Comment:
too many exes and also shouldn't be x but like the cross product symbol

Reply:
Fixed